### DISCUSSION TITLE
Is posibble to catch command from object?

### DISCUSSION THREAD

**Discussion by naufalkharits** [telegraf/telegraf#1782] [Q&A] [answered]
I modified the message.text from string to object.
```
bot.use((ctx, next) => {
  ctx.message.text = { origin: ctx.message.text, modified: ctx.message.text.toLowerCase() }

  return next(ctx)
})
```

I got error when catching a command. Is there any method to solve this?
```
bot.command("start", (ctx) => ctx.reply(`Hello, ${ctx.message.from.first_name}!`))
```

**Comment by MKRhere** [comment, marked as answer]
```TS
ctx.message.text = { origin: ctx....
```

Why would you do this? Telegraf relies on update objects being correct to route them. Use `ctx.state` to pass any custom data between middleware.

> Is there any method to solve this?

No, if you break Telegraf's core assumptions, it will break in unexpected ways. Don't do that.
